src/ast/mod.rs

```diff
@@ -733,6 +733,10 @@ pub enum Expr {
     ///
     /// See <https://docs.snowflake.com/en/sql-reference/constructs/where#joins-in-the-where-clause>.
     OuterJoin(Box<Expr>),
+    ColumnWithDataType {
+        columns: Box<Expr>,
+        data_type: DataType,
+    },
```

Comment:
I'm wondering if naming wise, 'ColumnWithDataType' sounds a bit too generic since it applies to a lot of unrelated scenarios. From the docs, it sounds incorrect as well to have the `columns: Box<Expr>`?
-  The actual column in `WHERE a = {b:INT}` is `a`, while `b` would be more of a function parameter
-  `b` given its a function parameter likely needs to be an `Ident` instead of an `Expr`

thinking roughly something along this line could be more descriptive
```rust
ParameterizedViewArg{
    name: Ident,
    data_type: DataType,
}
```

Reply:
Sure, I'll change it